### PR TITLE
fault_k8s to main branch merge

### PR DESCRIPTION
**Pull request by ArchanaLimaye**
# Problem Statement
fault_k8s to main branch merge

# Design
fault_k8s to main branch merge

# Coding
-  [ ] Coding conventions are followed and code is consistent

# Testing 
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Review Checklist 
- [ ] PR is self reviewed
- [ ] JIRA number/GitHub Issue added to PR
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: 
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N]
- [ ] Side effects on other features (deployment/upgrade)? [Y/N]
- [ ] Dependencies on other component(s)? [Y/N]
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by shaileshsvc — APPROVED** [review]
Approved.

File: ha/fault_tolerance/cluster_stop_monitor.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    def __init__(self):
+        """Init method"""
+        self._consumer = self._get_consumer()
+
+    def start(self):
+        """
+        Start to listen messages.
+        """
+        self._consumer.start()
```
Comment (Madhura-08) [inline ha/fault_tolerance/cluster_stop_monitor.py:41]: I think there is a check here for self._consumer. whether its None or not

File: ha/fault_tolerance/fault_tolerance_driver.py [hunk 2]
```diff
@@ -41,42 +37,29 @@ def __init__(self, wait_time=10):
...
-        except Exception as err:
-            Log.error(f'Failed to analyze the event: {err}')
-            return CONSUMER_STATUS.FAILED
-        return CONSUMER_STATUS.SUCCESS
+        self.node_fault_monitor.start()
+        self.cluster_stop_monitor.start()
 
     def poll(self):
-        """Contineously polls for message bus for k8s_event message type"""
+        Log.debug("FaultTolerance poll")
```
Comment (Madhura-08) [inline ha/fault_tolerance/fault_tolerance_driver.py:51]: similarly here there should be function docstring. Right?

File: ha/fault_tolerance/cluster_stop_monitor.py (hunk 1 above)
Comment (ArchanaLimaye) [reply ha/fault_tolerance/cluster_stop_monitor.py:41]: This was added in fault_monitor [ The derived class was used for getting consumerthere]

File: ha/fault_tolerance/fault_tolerance_driver.py (hunk 2 above)
Comment (ArchanaLimaye) [reply ha/fault_tolerance/fault_tolerance_driver.py:51]: Good catch. Looks like the docstring moved one line down

File: ha/fault_tolerance/cluster_stop_monitor.py (hunk 1 above)
Comment (Madhura-08) [reply ha/fault_tolerance/cluster_stop_monitor.py:41]: Got it. Thanks!

File: ha/fault_tolerance/fault_tolerance_driver.py (hunk 2 above)
Comment (Madhura-08) [reply ha/fault_tolerance/fault_tolerance_driver.py]: Ahh ok..In that PR also, this was missed. I checked only comment is there or not but not the alignment. Anyways, thanks for fixing.

**Review by Madhura-08 — APPROVED** [review]

**Review by akash2144 — APPROVED** [review]

**Review by mukhtar-inamdar — APPROVED** [review]

**Pull request merged**